core/src/main/scala/com/yahoo/maha/core/query/OffHeapRowList.scala

```diff
@@ -0,0 +1,194 @@
+package com.yahoo.maha.core.query
+
+import java.io.File
+import java.util.UUID
+import java.util.concurrent.atomic.AtomicLong
+
+import com.yahoo.maha.core.query.QueryRowList.ROW_COUNT_ALIAS
+import com.yahoo.maha.core.request.Parameter.RequestId
+import com.yahoo.maha.rocksdb.{RocksDBAccessor, RocksDBAccessorBuilder}
+import com.yahoo.maha.serde.LongSerDe
+import grizzled.slf4j.Logging
+import org.apache.commons.io.{FileUtils, FilenameUtils}
+
+import scala.collection.mutable.ArrayBuffer
+import scala.util.Try
+
+/*
+    Created by pranavbhole on 2/3/20
+*/
+case class OffHeapRowListConfig(tempStoragePathOption:Option[String], inMemRowCountThreshold: Int = 10000)
+
+case class OffHeapRowList(query: Query, config: OffHeapRowListConfig) extends QueryRowList with AutoCloseable with Logging {
+
+  private[this] val tempStoragePath = OffHeapRowList.getAndValidateTempStorage(config.tempStoragePathOption)
+
+  private[this] var rocksDBAccessorOption: Option[RocksDBAccessor[Long, Row]] = None
+
+  private[this] val atomicLongRowCount = new AtomicLong(-1)
```

Comment:
Does this need to be atomiclong?  Are multiple threads going to be writing to the row list concurrently?

Reply:
fixed, no need of atomic